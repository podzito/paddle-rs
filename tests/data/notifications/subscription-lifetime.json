{
  "data": {
    "id": "ntf_01j6ffntraahbc78qjaqpvta49",
    "type": "transaction.completed",
    "status": "not_attempted",
    "payload": {
      "data": {
        "id": "txn_01j6ffmn625kgf1qj2rec51mjy",
        "items": [
          {
            "price": {
              "id": "pri_01j6eys6jwj3msa87b7nknd6jc",
              "name": "PodZito Pro - Lifetime",
              "type": "standard",
              "status": "active",
              "quantity": { "maximum": 1, "minimum": 1 },
              "tax_mode": "account_setting",
              "created_at": "2024-08-29T11:36:54.620513Z",
              "product_id": "pro_01j52z87e5sdt3p60mmtcavpgt",
              "unit_price": { "amount": "19900", "currency_code": "USD" },
              "updated_at": "2024-08-29T11:36:54.620513Z",
              "custom_data": null,
              "description": "PodZito Pro - Lifetime",
              "trial_period": null,
              "billing_cycle": null,
              "unit_price_overrides": []
            },
            "price_id": "pri_01j6eys6jwj3msa87b7nknd6jc",
            "quantity": 1,
            "proration": null
          }
        ],
        "origin": "web",
        "status": "completed",
        "details": {
          "totals": {
            "fee": "1045",
            "tax": "3317",
            "total": "19900",
            "credit": "0",
            "balance": "0",
            "discount": "0",
            "earnings": "15538",
            "subtotal": "16583",
            "grand_total": "19900",
            "currency_code": "USD",
            "credit_to_balance": "0"
          },
          "line_items": [
            {
              "id": "txnitm_01j6ffmyez4rs05tevrbm8dmjw",
              "totals": { "tax": "3317", "total": "19900", "discount": "0", "subtotal": "16583" },
              "item_id": null,
              "product": {
                "id": "pro_01j52z87e5sdt3p60mmtcavpgt",
                "name": "PodZito",
                "type": "standard",
                "status": "active",
                "image_url": null,
                "created_at": "2024-08-12T09:38:32.005Z",
                "updated_at": "2024-08-12T09:38:32.005Z",
                "custom_data": null,
                "description": "Podzito",
                "tax_category": "standard"
              },
              "price_id": "pri_01j6eys6jwj3msa87b7nknd6jc",
              "quantity": 1,
              "tax_rate": "0.2",
              "unit_totals": { "tax": "3317", "total": "19900", "discount": "0", "subtotal": "16583" }
            }
          ],
          "payout_totals": {
            "fee": "1045",
            "tax": "3317",
            "total": "19900",
            "credit": "0",
            "balance": "0",
            "discount": "0",
            "earnings": "15538",
            "fee_rate": "0.05",
            "subtotal": "16583",
            "grand_total": "19900",
            "currency_code": "USD",
            "exchange_rate": "1",
            "credit_to_balance": "0"
          },
          "tax_rates_used": [
            { "totals": { "tax": "3317", "total": "19900", "discount": "0", "subtotal": "16583" }, "tax_rate": "0.2" }
          ],
          "adjusted_totals": {
            "fee": "1045",
            "tax": "3317",
            "total": "19900",
            "earnings": "15538",
            "subtotal": "16583",
            "grand_total": "19900",
            "currency_code": "USD"
          }
        },
        "checkout": { "url": "https://www.podzito.com?_ptxn=txn_01j6ffmn625kgf1qj2rec51mjy" },
        "payments": [
          {
            "amount": "19900",
            "status": "captured",
            "created_at": "2024-08-29T16:32:06.14845Z",
            "error_code": null,
            "captured_at": "2024-08-29T16:32:08.170941Z",
            "method_details": {
              "card": {
                "type": "visa",
                "last4": "5556",
                "expiry_year": 2029,
                "expiry_month": 9,
                "cardholder_name": "AAA"
              },
              "type": "card"
            },
            "payment_method_id": "paymtd_01j6ffnpzkq7sfzayktbwq44v9",
            "payment_attempt_id": "77fb0f2c-5e4f-47d9-9bf5-92b51afabf98",
            "stored_payment_method_id": "b96e1131-a9dd-43ce-b186-5406f688fe6a"
          }
        ],
        "billed_at": "2024-08-29T16:32:08.561157Z",
        "address_id": "add_01j6ffmy5yggg1vxtr7cwj5439",
        "created_at": "2024-08-29T16:31:31.612753Z",
        "invoice_id": "inv_01j6ffnsqp61r845d8s30d5nqh",
        "updated_at": "2024-08-29T16:32:09.431647461Z",
        "business_id": null,
        "custom_data": null,
        "customer_id": "ctm_01j535fawggst1gm3qsyanc0mb",
        "discount_id": null,
        "receipt_data": null,
        "currency_code": "USD",
        "billing_period": null,
        "invoice_number": "8326-10027",
        "billing_details": null,
        "collection_mode": "automatic",
        "subscription_id": null
      },
      "event_id": "evt_01j6ffntn1m7vt4jpttvcm8y62",
      "event_type": "transaction.completed",
      "occurred_at": "2024-08-29T16:32:09.889866Z",
      "notification_id": "ntf_01j6ffntraahbc78qjaqpvta49"
    },
    "occurred_at": "2024-08-29T16:32:09.889866Z",
    "delivered_at": null,
    "replayed_at": null,
    "origin": "event",
    "last_attempt_at": null,
    "retry_at": null,
    "times_attempted": 0,
    "notification_setting_id": "ntfset_01j5akfpdszj2r2prnsjbf0ytt"
  },
  "meta": { "request_id": "78a540bb-128e-4549-9a6a-0fa53fcce0d2" }
}

