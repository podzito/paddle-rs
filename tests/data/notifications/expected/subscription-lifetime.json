{
  "event_id": "evt_01j6ffntn1m7vt4jpttvcm8y62",
  "notification_id": "ntf_01j6ffntraahbc78qjaqpvta49",
  "event": {
    "TransactionCompleted": {
      "id": "txn_01j6ffmn625kgf1qj2rec51mjy",
      "customer_id": "ctm_01j535fawggst1gm3qsyanc0mb",
      "items": [
        {
          "price": {
            "billing_cycle": null,
            "created_at": "2024-08-29T11:36:54.620513Z",
            "custom_data": null,
            "description": "PodZito Pro - Lifetime",
            "id": "pri_01j6eys6jwj3msa87b7nknd6jc",
            "import_meta": null,
            "name": "PodZito Pro - Lifetime",
            "product_id": "pro_01j52z87e5sdt3p60mmtcavpgt",
            "quantity": { "maximum": 1, "minimum": 1 },
            "status": "active",
            "tax_mode": "account_setting",
            "trial_period": null,
            "type": "standard",
            "unit_price": { "amount": "19900", "currency_code": "USD" },
            "updated_at": "2024-08-29T11:36:54.620513Z"
          },
          "product": null,
          "created_at": null,
          "next_billed_at": null,
          "previously_billed_at": null,
          "price_id": "pri_01j6eys6jwj3msa87b7nknd6jc",
          "quantity": 1,
          "recurring": null,
          "status": null,
          "trial_dates": null,
          "updated_at": null
        }
      ],
      "custom_data": null
    }
  },
  "occurred_at": "2024-08-29T16:32:09.889866Z"
}
